File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/string/TrimToTest.java
package com.fincity.nocode.kirun.engine.function.system.string;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

class TrimToTest {

	@Test
	void test() {

		String s1 = " THIScompatY IS A NOcoDE plATFNORM";

		TrimTo trimed = new TrimTo();

		StepVerifier
				.create(trimed.execute(new ReactiveFunctionExecutionParameters(new KIRunReactiveFunctionRepository(),
						new KIRunReactiveSchemaRepository()).setArguments(
								Map.of(TrimTo.PARAMETER_STRING_NAME,
										new JsonPrimitive(s1), TrimTo.PARAMETER_LENGTH_NAME, new JsonPrimitive(14)))))
				.expectNextMatches(fo -> fo.next().getResult().get(TrimTo.EVENT_RESULT_NAME).getAsString()
						.equals(" THIScompatY I"))
				.verifyComplete();
	}

	@Test
	void test2() {

		String s1 = " THIScompatY IS A NOcoDE plATFNORM";

		TrimTo trimed = new TrimTo();

		StepVerifier
				.create(trimed.execute(new ReactiveFunctionExecutionParameters(new KIRunReactiveFunctionRepository(),
						new KIRunReactiveSchemaRepository()).setArguments(
								Map.of(TrimTo.PARAMETER_STRING_NAME,
										new JsonPrimitive(s1), TrimTo.PARAMETER_LENGTH_NAME, new JsonPrimitive(0)))))
				.expectNextMatches(fo -> fo.next().getResult().get(TrimTo.EVENT_RESULT_NAME).getAsString()
						.equals(""))
				.verifyComplete();
	}
}

File: kirun-js/__tests__/engine/function/system/string/TrimToTest.ts
import { TrimTo } from '../../../../../src/engine/function/system/string/TrimTo';
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';

const trim: TrimTo = new TrimTo();

test('Trim to test1 ', async () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map<string, string | number>([
            [TrimTo.PARAMETER_STRING_NAME, ' THIScompatY IS A NOcoDE plATFNORM'],
            [TrimTo.PARAMETER_LENGTH_NAME, 14],
        ]),
    );

    expect(
        (await trim.execute(fep)).allResults()[0].getResult().get(TrimTo.EVENT_RESULT_NAME),
    ).toBe(' THIScompatY I');
});

test('Trim to test2 ', async () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map<string, string | number>([
            [TrimTo.PARAMETER_STRING_NAME, ' THIScompatY IS A NOcoDE plATFNORM'],
            [TrimTo.PARAMETER_LENGTH_NAME, 0],
        ]),
    );

    expect(
        (await trim.execute(fep)).allResults()[0].getResult().get(TrimTo.EVENT_RESULT_NAME),
    ).toBe('');
});
